docs/reference/cryptographic-overview.md

```diff
@@ -0,0 +1,47 @@
+# GitHub runner cryptographic overview
+This document provides an overview of the cryptographic technologies used in the GitHub runner charm, including encryption/decryption, hashing, and digital signatures.
+
+## Overall Description of Cryptographic Technology Use
+The GitHub runner charm uses various cryptographic technologies to ensure secure communication and data integrity. The following sections describe each of the components.
+### TLS
+The GitHub runner communicates with GitHub to receive information about the workflows to be executed, retrieve the repositories config and send back workflow logs to the user. This communication uses [urllib3](https://urllib3.readthedocs.io/en/stable/) under the hood using TLS 1.3.
+
+In OpenStack mode, the runner creates virtual machines to run the workload in OpenStack. The GitHub runner charm interacts with the OpenStack API to create and manage virtual machines. These interactions are secured via TLS. The communication between these virtual machines and runner is done via SSH and secured by 256 byte keypairs.
```

Comment:
Nit: would the following info be helpful?

The SSH key is created by OpenStack, which uses ssh PEM keys (OpenSSL).
```
$ ssh-keygen -l -f ./<keyfile>
2048 SHA256:w0gf4CC2xY3X9z+WRwT1xh4osy1j25hsm3fCg7bUOqk no comment (RSA)
```